File: rea2/app/src/main/java/com/jang/assistales/model/ProjectBean.java
package com.jang.assistales.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity
public class ProjectBean implements Serializable {
    @PrimaryKey(autoGenerate = true)
    private long id_project;

    private String name;
    private String description;
    private String storyline;

    @Override
    public String toString() {
        return name;
    }

    //--------------------------Constructor---------------------------//
    public ProjectBean(String name) {
        this.name = name;
    }

    //--------------------------Getter/Setter-------------------------//


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStoryline() {
        return storyline;
    }

    public void setStoryline(String storyline) {
        this.storyline = storyline;
    }
}

File: rea5/app/src/main/java/com/jang/assistales/model/ProjectBean.kt
package com.jang.assistales.model

import androidx.room.*

@Entity(
    tableName = "project"
)
class ProjectBean (
    val name: String,
    @PrimaryKey(autoGenerate = true)
    val id_project: Long = 0,
    val description: String? = null,
    val storyline: String? = null

) {
    override fun toString(): String {
        return name
    }


    @Dao
    interface ProjectDao {
        @Query("SELECT * FROM project ORDER BY name ASC")
        fun getAll(): List<ProjectBean>

        @Insert
        fun insert(project: ProjectBean): Long

        @Update
        fun update(project: ProjectBean)

        @Delete
        fun delete(project: ProjectBean)
    }
}